### PR TITLE
Development

### PR DESCRIPTION
**Pull request by n-maido**
## Backend
 - Established communication between `django` and `sql` using `graphql` (#7)
    
Screen Shot 2021-08-06 at 12 05 40 AM
(testing addEvent, eventById, allEvents queries on graphql endpoint)

## Frontend
 - Implemented a login and registration page ( #4)
    - `frontend/lib/main.dart`
    
login

 - Implemented a browse events page that displays event data as a list of cards (#5)
    - `frontend/lib/screens/browse_events.dart`
    - `frontend/lib/widgets/registration_button.dart`
    
browse

 - Implemented a hamburger menu and basic routing (#6)
   - `frontend/lib/widgets/nav_drawer.dart`
    
   menu

### CODE REVIEW

**Comment by n-maido** [comment]
@YenyuHuang @manlalaro1 @shivichaubey

File: backend/users/models.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-    REQUIRED_FIELDS = ['username']
-    username = models.CharField(blank=True, max_length=254,)
-    email = models.EmailField(blank=False, max_length=254, unique=True, verbose_name='email address')
\ No newline at end of file
+class Event(models.Model):
+    name = models.CharField(max_length=100)
+    date = models.DateTimeField()
+    tag = models.CharField(max_length=100)
+    organizer = models.CharField(max_length=100)
+    location = models.CharField(max_length=100)
```
Comment (yashika51) [inline backend/users/models.py:10]: Would having a  `description` as well make sense? You can think and decide about it.

File: frontend/lib/screens/browse_events.dart [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        "location": "1234 W Sample St, Vancouver, BC"
+      },
+      {
+        "name": 'Martha\'s Birthday Bash',
+        "date": "2016-07-20T12:00:15+00:00",
+        "tag": "Meeting",
+        "organizer": "Martha's Family",
+        "location": "1234 W Sample St, Vancouver, BC"
+      }
+    ];
```
Comment (yashika51) [inline frontend/lib/screens/browse_events.dart:27]: Could this have been a `yaml` file instead?

File: frontend/lib/widgets/nav_drawer.dart [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        break;
+      // case 3:
+      //   Navigator.of(context).push(MaterialPageRoute(builder: (context) => YourPageName()));
+      //   break;
+      // case 4:
+      //   Navigator.of(context).push(MaterialPageRoute(builder: (context) => YourPageName()));
+      //   break;
+      // case 5:
+      //   Navigator.of(context).push(MaterialPageRoute(builder: (context) => YourPageName()));
+      //   break;
```
Comment (yashika51) [inline frontend/lib/widgets/nav_drawer.dart:132]: Commented code shouldn't be in the master branch :)

**Review by yashika51 — CHANGES_REQUESTED** [review]

File: frontend/lib/widgets/register_button.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    return ElevatedButton(
+      onPressed: null, 
+       //style: ButtonStyle(padding: EdgeInsets.all(0.0), ),
+      child: const Text('Register', style: TextStyle(fontSize: 11)) );
+  }
+
+  // void _registerPressed() {
+    //   Navigator.push(context, MaterialPageRoute(builder: (context) => BrowseEvents()));
+    // }
+}
```
Comment (yashika51) [inline frontend/lib/widgets/register_button.dart:20]: No commented code

File: frontend/macos/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme [hunk 5]
```diff
@@ -15,7 +15,7 @@
             <BuildableReference
                BuildableIdentifier = "primary"
                BlueprintIdentifier = "33CC10EC2044A3C60003C045"
```
Comment (yashika51) [inline frontend/macos/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:17]: Not sure what this is, if this is a key this should be in `.env` file instead.

**Review by yashika51 — CHANGES_REQUESTED** [review]

File: frontend/lib/screens/browse_events.dart (hunk 2 above)
Comment (n-maido) [reply frontend/lib/screens/browse_events.dart]: Moved this to a json file in commit 029e4c6f432a732625dbf521275750643f5e2128

File: frontend/lib/widgets/nav_drawer.dart (hunk 3 above)
Comment (n-maido) [reply frontend/lib/widgets/nav_drawer.dart:132]: Uncommented this block in commit 029e4c6f432a732625dbf521275750643f5e2128

File: frontend/lib/widgets/register_button.dart (hunk 4 above)
Comment (n-maido) [reply frontend/lib/widgets/register_button.dart]: Removed this in commit 029e4c6f432a732625dbf521275750643f5e2128

File: backend/users/models.py (hunk 1 above)
Comment (manlalaro1) [reply backend/users/models.py:10]: Added description data type in commit 5c07a74f41277987ca66d029f8b1b310e674a92a

File: frontend/macos/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (hunk 5 above)
Comment (YenyuHuang) [reply frontend/macos/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme:17]: Added sensitive keys: commit:  2a6e78675aafe70458f5179aa564457e05e65eec
specifically macOS BuildableReference Identifiers in .env file

1. created .env file in frontend folder
2. added *.env in both .gitignore
3. import package flutter_dotenv int main.dart
4. added DotEnv variable  in main.dart to load the key
5. flutter pub get to get flutter_dotenv dependency (flutter_dotenv^5.0.0)

result: app still functions

**Pull request merged**